include/Flow.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "NetworkDevice.hpp"
+
+#include <cstdint>
+#include <memory>
+#include <queue>
+
+class Flow {
+public:
+    explicit Flow(const std::vector<std::shared_ptr<NetworkDevice>>& path, const std::vector<std::uint32_t>& distances_ns);
+
+    void Send();
+
+    std::uint32_t id;
+    std::vector<std::shared_ptr<NetworkDevice>> path{};
+    std::vector<std::uint32_t> distances_ns{};
+
+    double interval_ns;
```

Comment:
What is `interval_ns` for?

Reply:
I renamed it to `packet_generation_interval_ns`. It is a event delay time in `Flow::Send()`. 

A new packet is enqueued every `packet_generation_interval_ns` nanoseconds.

Reply:
Could you, please, draw a schematics of the network device you have in mind?

Reply:
![image](https://github.com/user-attachments/assets/ccb7e9a4-0ddd-47e1-97b4-0f343f1faaa5)

Could you please explain why the delta is needed?

Reply:
Sorry, this sketch was meant to be here https://github.com/Cloud-Storage-Team/algnet/pull/33

Reply:
> Could you please explain why the delta is needed?

This is the exact formula which account that `t_out-t_in` may not be multiple of `packet_processing_delay`. Assuming that `packet_processing_delay` is small compared to other delays, `delta` can be neglected.

Reply:
Actually, the formula that use https://github.com/Cloud-Storage-Team/algnet/pull/34#discussion_r1895677184 takes into account this `delta` naturally: it sits in `predicted_completion_time - now`, which may not be multiple of the number of packets.